extensions/aproc/proc/access/manager.py

```diff
@@ -116,22 +122,71 @@ def prepare(href: str):
         """
         storage = AccessManager.resolve_storage(href)
 
-        # If the storage is nit local, pull it
-        if storage.type != "file":
-            dst = os.path.join(AccessManager.tmp_dir, os.path.basename(href))
-            storage.pull(href, dst)
-            return dst
+        # If the storage is not local, pull it
+        if not storage.is_local:
+            if dst is None:
+                dst = os.path.join(AccessManager.tmp_dir, os.path.basename(href))
 
-        return href
+            AccessManager.pull(href, dst)
+            try:
+                yield dst
+            finally:
+                AccessManager.clean(dst)  # !DELETE!
+        else:
+            yield href
 
     @staticmethod
-    def zip(href: str, zip_path: str):
-        # For all storages but FileStorage, files need to be pulled before being processed
-        href = AccessManager.prepare(href)
+    @contextmanager
+    def make_local_list(href_list: list[str], dst_list: list[str | None] | None = None):
+        """Prepare a list of files to make them available locally for further processing.
+           Once used, the file is deleted if it has been pulled
+        """
+        # Check that the input lists match each other length
+        if dst_list is not None and len(href_list) != len(dst_list):
+            raise ValueError("Input href and dst must have the same length")
+        if dst_list is None:
+            dst_list = [None for _ in href_list]
+
+        # For each of the input pair of (href, dst), check if the corresponding storage is local
+        # If not, pull it, store dst and tag the iteration as pulled. Otherwise, store href and tag as not pulled.
+        # Once all local, will yield the list of local paths
+        # Cleanup will only remove the files that were pulled
+        local_href_list = []
+        was_pulled = []
+        try:
+            for href, dst in zip(href_list, dst_list):
+                storage = AccessManager.resolve_storage(href)
+
+                if not storage.is_local:
+                    if dst is None:
+                        dst = os.path.join(AccessManager.tmp_dir, os.path.basename(href))
+
+                    AccessManager.pull(href, dst)
+                    local_href_list.append(dst)
+                    was_pulled.append(True)
+                else:
+                    local_href_list.append(href)
+                    was_pulled.append(False)
+            yield local_href_list
+        finally:
+            for pulled, local_href in zip(was_pulled, local_href_list):
+                if pulled:
+                    AccessManager.clean(local_href)  # !DELETE!
```

Comment:
Add comment, e.g.:

`# Only pulled files (files downloaded by this process) are deleted, as a clean up procedure.`